pinecone/control/pinecone.py

```diff
@@ -561,15 +575,28 @@ def configure_index(
 
         """
         api_instance = self.index_api
-        config_args: Dict[str, Any] = {}
+
+        if deletion_protection is None:
+            description = self.describe_index(name=name)
+            dp = DeletionProtection(description.deletion_protection)
+        elif deletion_protection in ["enabled", "disabled"]:
+            dp = DeletionProtection(deletion_protection)
+        else:
+            raise ValueError("deletion_protection must be either 'enabled' or 'disabled'")
+
+        pod_config_args: Dict[str, Any] = {}
         if pod_type:
-            config_args.update(pod_type=pod_type)
+            pod_config_args.update(pod_type=pod_type)
         if replicas:
-            config_args.update(replicas=replicas)
-        configure_index_request = ConfigureIndexRequest(
-            spec=ConfigureIndexRequestSpec(pod=ConfigureIndexRequestSpecPod(**config_args))
-        )
-        api_instance.configure_index(name, configure_index_request=configure_index_request)
+            pod_config_args.update(replicas=replicas)
+
+        if pod_config_args != {}:
+            spec = ConfigureIndexRequestSpec(pod=ConfigureIndexRequestSpecPod(**pod_config_args))
```

Comment:
I hate the arrangement of this code. Blame the black formatter. I need to update the configs to shorten the line length to something more reasonable.